compatibility_lib/compatibility_lib/compatibility_store.py

```diff
@@ -394,16 +353,22 @@ def save_compatibility_statuses(
         rows = [self._compatibility_status_to_row(s) for s in
                 compatibility_statuses]
 
-        self_rows = [r for r in rows if 'install_name' in r]
-        pair_rows = [r for r in rows if 'install_name' not in r]
-        if self_rows:
-            self._client.insert_rows(
-                self._self_table,
-                self_rows)
-        if pair_rows:
-            self._client.insert_rows(
-                self._pairwise_table,
-                pair_rows)
+        self_rows = [r for r in rows if len(r) == 5]
+        pair_rows = [r for r in rows if len(r) == 6]
+
+        with closing(self.connect()) as conn:
+            with closing(conn.cursor()) as cursor:
+                if self_rows:
+                    self_sql = ('REPLACE INTO self_compatibility_status '
+                                'values (%s, %s, %s, %s, %s)')
+                    cursor.executemany(self_sql, self_rows)
+                    conn.commit()
```

Comment:
Maybe you should move the commit to the same level as `if self_rows` and only have one?

Reply:
Done